internal/wrappers/headers/kuzzle.hpp

```diff
@@ -0,0 +1,155 @@
+    std::string getMessage() const;
+  };
+
+  struct BadRequestException : KuzzleException {
+    BadRequestException(const std::string& message="Bad Request Exception")
+      : KuzzleException(400, message) {};
+    BadRequestException(const BadRequestException& bre) : KuzzleException(bre.status, bre.getMessage()) {}
+  };
+  struct ForbiddenException: KuzzleException {
+    ForbiddenException(const std::string& message="Forbidden Exception")
+      : KuzzleException(403, message) {};
+    ForbiddenException(const ForbiddenException& fe) : KuzzleException(fe.status, fe.getMessage()) {}
+  };
+  struct GatewayTimeoutException: KuzzleException {
+    GatewayTimeoutException(const std::string& message="Gateway Timeout Exception")
+      : KuzzleException(504, message) {};
+    GatewayTimeoutException(const GatewayTimeoutException& gte) : KuzzleException(gte.status, gte.getMessage()) {}
+  };
+  struct InternalException: KuzzleException {
+    InternalException(const std::string& message="Internal Exception")
+      : KuzzleException(500, message) {};
+    InternalException(const InternalException& ie) : KuzzleException(ie.status, ie.getMessage()) {}
+  };
+  struct NotFoundException: KuzzleException {
+    NotFoundException(const std::string& message="Not Found Exception")
+      : KuzzleException(404, message) {};
+    NotFoundException(const NotFoundException& nfe) : KuzzleException(nfe.status, nfe.getMessage()) {}
+  };
+  struct PartialException: KuzzleException {
+    PartialException(const std::string& message="Partial Exception")
+      : KuzzleException(206, message) {};
+    PartialException(const PartialException& pe) : KuzzleException(pe.status, pe.getMessage()) {}
+  };
+  struct PreconditionException: KuzzleException {
+    PreconditionException(const std::string& message="Precondition Exception")
+      : KuzzleException(412, message) {};
+    PreconditionException(const PreconditionException& pe) : KuzzleException(pe.status, pe.getMessage()) {}
+  };
+  struct ServiceUnavailableException: KuzzleException {
+    ServiceUnavailableException(const std::string& message="Service Unavailable Exception")
+      : KuzzleException(503, message) {};
+    ServiceUnavailableException(const ServiceUnavailableException& sue) : KuzzleException(sue.status, sue.getMessage()) {}
+  };
+  struct SizeLimitException: KuzzleException {
+    SizeLimitException(const std::string& message="Size Limit Exception")
+      : KuzzleException(413, message) {};
+    SizeLimitException(const SizeLimitException& sle) : KuzzleException(sle.status, sle.getMessage()) {}
+  };
+  struct UnauthorizedException : KuzzleException {
+    UnauthorizedException(const std::string& message="Unauthorized Exception")
+     : KuzzleException(401, message) {}
+    UnauthorizedException(const UnauthorizedException& ue) : KuzzleException(ue.status, ue.getMessage()) {}
+  };
+
+  template <class T>
+  void throwExceptionFromStatus(T result) {
+    if (result.status == 206)
```

Comment:
nitpicking: a switch..case statement may be more readable

not a nitpicking: what exception do we raise if a status value is different from the ones listed?